app/views/grids/_notes.html.erb

```diff
@@ -67,9 +67,17 @@
 <% if type == "questions" || tagname.include?('question:') %>
   <p><a href='/post?tags=question:<%= tagname %>,<%= tagname %>&template=question&title=How%20do%20I...&redirect=question' class='btn btn-primary add-activity'>Ask a question</a> &nbsp;or <a href='/subscribe/tag/question:<%= tagname %>'>help answer future questions<span class='hidden-sm hidden-xs'> on this topic</span></a></p>
 <% elsif type == "activity" || tagname.include?('activity:') %>
-  <p><a href='/post?tags=activity:<%= tagname %>,<%= tagname %>,seeks:replications&title=How%20to%20do%20X' class='btn btn-primary add-activity'>Add an activity</a> &nbsp;or <a href='/post?tags=<%= tagname %>,question:<%= tagname %>,request:activity&template=question&title=How%20do%20I...&redirect=question' class='request-activity'>request an activity<span class='hidden-xs hidden-sm'> guide you don't see listed</span></a></p>
+  <p id = "buttons-<%= randomSeed %>"><a href='/post?tags=activity:<%= tagname %>,<%= tagname %>,seeks:replications&title=How%20to%20do%20X' class='btn btn-primary add-activity'>Add an activity</a> &nbsp;or <a href='/post?tags=<%= tagname %>,question:<%= tagname %>,request:activity&template=question&title=How%20do%20I...&redirect=question' class='request-activity'>request an activity<span class='hidden-xs hidden-sm'> guide you don't see listed</span></a></p>
   <p><i>Activities should include a materials list, costs and a step-by-step guide to construction with photos. Learn what <a href="https://publiclab.org/notes/warren/09-17-2016/what-makes-a-good-activity">makes a good activity here</a>.</i>
   </p>
+  <script>
+    $(document).ready(function() {      
+      var href1 = '/post?tags=activity:<%= tagname %>,<%= tagname %>,seeks:replications&title=How%20to%20do%20X';
+      var href2 = '/post?tags=<%= tagname %>,question:<%= tagname %>,request:activity&template=question&title=How%20do%20I...&redirect=question';
+      gridLinkModal('#buttons-<%= randomSeed %>', href1);
+      gridLinkModal('#buttons-<%= randomSeed %>', href2);
```

Comment:
Hmm, how do these two trigger 2 different modals? I don't see a unique identifier... maybe just clarify this? Thanks!

Reply:
Correct me if I'm wrong, but I think he's triggering the same modal, but with different links to load after the login has completed.

Reply:
Yep thats exactly right. The href variable is used to send a getjson request to change the request so that the user is redirected appropriately.

Reply:
OK, i'm sorry to be slowing this down, but (although it's been a long day and maybe a fresh brain would work better) I'm still having trouble understanding each step in this code. I'm wondering if some comments inline would help explain what each does - although it's beautifully coded, and compact code is really nice, I really want to ensure it's easy to read for future coders. Do you think you could add some extra notes on what's going on here so that future coders don't break it or get confused when reading it? 

Thank you! There've often been moments when i tried to read my own code from 2 years ago, and been a bit mystified. So I think this could be pretty worthwhile. I appreciate your time!

Reply:
One way to make the code very readable (alongside some comments) is to use pretty descriptive function and variable names. So `gridLinkModal` could be something like `linkButtonToLoginModal()` and, for example, `current_user` if I read it right could be named `isLoggedIn`? What do you think?

Reply:
We can actually remove all of that if #4264 is merged!